docs/tutorials/porting.md

```diff
@@ -0,0 +1,225 @@
+## Mbed OS porting guide
+
+This guide walks you through the steps required to port Mbed OS onto a derivative target of any one of the Mbed OS Enabled targets. If you need to port the MCU to Mbed OS, as well, along with a custom board based on the MCU, then you can follow our [more detailed porting guide](docs/development/reference/porting-targets.html). 
+
+Mbed Cloud Client provides reference implementation for three Mbed Enabled targets, the K64F, NUCLEO_F429ZI and the UBLOX_EVK_ODIN_W2. You do not need to port either Mbed OS or the Cloud Client if you are using one of these three boards.
+
+### Summary of steps
+
+1. Add your target to `targets.json`
+   1. Identify the your target’s features.
+   1. Identify any existing parent targets.
+   1. Identify macros required for your target's compilation.
+   1. Identify supported compilers and the Mbed OS release version.
+   1. Add the target entry in `targets.json`.
+   1. Verify that you can now pass your target to `mbed compile`.
+1. Add your target port.
+   1. Create the directory structure.
+   1. Add an entry for the target in `mbed_rtx.h`.
+   1. Add startup code and CMSIS headers, NVIC headers and **all** relevant drivers from CMSIS specifications.
+   1. Implement APIs, HAL, the system clock configuration and any other additional clocks.
+   1. Add toolchain-specific linker descriptions.
+1. Add peripherals and pin names for the target.
+1. Compile with CLI on a supported compiler.
+
+### Identify target properties
+
+You can identify our sample target by the following:
+
+- The name of the target for builds is `MY_BOARD_1`.
+- The vendor is "MY_VENDOR", the device family is "MY_FAMILY" and the device on the board is "MY_DEVICE".
+- A Cortex-M4F based core.
+- Porting for Mbed OS v5.x, which requires support for Arm Compiler 5, GCC Arm Embedded and IAR EWARM.
+- Has a serial (UART) interface connected to the CMSIS-DAP implementation.
+- Requires the macro `CPU_DEVICE_1` to compile the vendor-provided HAL for the device mounted on the board.
+- Requires the macro `FAMILY_MY_FAMILY` to compile the vendor-provided HAL for the family of the device.
+
+All of these requirements directly map to relevant tags in the `targets.json` entry for our target. This is shown in step 3 below.
```

Comment:
Note: We don't have a step 3 anymore. What does this reference?

Reply:
The following step.